[issue]: error when run build script

When the script build is executed the prompt return a dependecies error:

```bash
> tsc -p tsconfig.json && rollup -c rollup.config.js

node_modules/@types/jest/index.d.ts:484:51 - error TS2307: Cannot find module 'jest-diff'.

484             diff(a: any, b: any, options?: import("jest-diff").DiffOptions): string | null;
                                                      ~~~~~~~~~~~

node_modules/@types/jest/index.d.ts:538:44 - error TS2307: Cannot find module 'pretty-format'.

538     type SnapshotSerializerPlugin = import('pretty-format').Plugin;
                                               ~~~~~~~~~~~~~~~

```

--- a/package.json
+++ b/package.json
@@ -8,15 +8,16 @@
   "license": "MIT",
   "husky": {
     "hooks": {
-      "pre-commit": "yarn run test && lint-staged"
+      "pre-commit": "yarn run test && lint-staged",
+      "pre-push": "yarn run build"
     }
   },
   "files": [
     "build"
   ],
   "scripts": {
-    "test": "jest",
-    "test:watch": "yarn run test -- --watch",
+    "test": "jest --coverage",
+    "test:watch": "jest --watch --coverage",
     "prebuild": "rm -rf dist/ && rm -rf build/",
     "build": "tsc -p tsconfig.json && rollup -c rollup.config.js",
     "postbuild": "cp dist/*.d.ts build/"
@@ -44,7 +45,22 @@
     "transform": {
       ".ts": "ts-jest"
     },
-    "testRegex": ".*\\.spec\\.ts$"
+    "testRegex": ".*\\.spec\\.ts$",
+    "coveragePathIgnorePatterns": [
+      "/node_modules/",
+      "/test/"
+    ],
+    "coverageThreshold": {
+      "global": {
+        "branches": 90,
+        "functions": 95,
+        "lines": 95,
+        "statements": 95
+      }
+    },
+    "collectCoverageFrom": [
+      "src/*.{js,ts}"
+    ]
   },
   "lint-staged": {
     "src/**/*.ts": [

--- a/package.json
+++ b/package.json
@@ -40,6 +40,7 @@
       "ts",
       "js"
     ],
+    "roots": ["<rootDir>/src"],
     "transform": {
       ".ts": "ts-jest"
     },

--- a/tsconfig.json
+++ b/tsconfig.json
@@ -39,7 +39,7 @@
     // "noFallthroughCasesInSwitch": true,    /* Report errors for fallthrough cases in switch statement. */
 
     /* Module Resolution Options */
-    // "moduleResolution": "node",            /* Specify module resolution strategy: 'node' (Node.js) or 'classic' (TypeScript pre-1.6). */
+    "moduleResolution": "node",            /* Specify module resolution strategy: 'node' (Node.js) or 'classic' (TypeScript pre-1.6). */
     // "baseUrl": "./",                       /* Base directory to resolve non-absolute module names. */
     // "paths": {},                           /* A series of entries which re-map imports to lookup locations relative to the 'baseUrl'. */
     // "rootDirs": [],                        /* List of root folders whose combined content represents the structure of the project at runtime. */

